clippy_lints/src/loops.rs

```diff
@@ -1865,6 +1893,29 @@ fn check_for_loop_over_map_kv<'tcx>(
     }
 }
 
+fn check_for_single_element_loop<'tcx>(cx: &LateContext<'tcx>, pat: &'tcx Pat<'_>, arg: &'tcx Expr<'_>) {
+    if_chain! {
+        if let ExprKind::AddrOf(BorrowKind::Ref, _, ref expr) = arg.kind;
+        if let PatKind::Binding(.., target, _) = pat.kind;
+        if let ExprKind::Array(ref expr_list) = expr.kind;
+        if expr_list.len() == 1;
+        if let ExprKind::Path(ref list_item) = expr_list[0].kind;
+        if let Some(list_tem_name) = single_segment_path(list_item).map(|ps| ps.ident.name);
+
+        then {
+            span_lint_and_sugg(
+                cx,
+                SINGLE_ELEMENT_LOOP,
+                expr.span,
```

Comment:
You should use the span from the start of the for loop to the end of `expr`. You can do this, by also passing the `expr` from the `check_for_loop` function and then use the span `for_expr.span.with_hi(expr.span.hi())`.